Improve focusable elements

MartijnCuppens opened the pull request:
- `<details>` is not focusable, `<summary>` is
- Use js properties instead of relying on attributes

zellwk commented:
@MartijnCuppens I'm not sure this make sense. I tested this on https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary. But `<summary>` is the one that was focusable, not `<details>`.

MartijnCuppens commented:
@zellwk, I've rebased the branch. 

> But `<summary>` is the one that was focusable, not `<details>`.

Yeah, that's exactly what I've changed in this PR.

I noticed you have applied the fix for negative tabindices, but you can just rely on the `tabIndex` property. I've also changed that in this PR.

zellwk commented:
🤦 What the hell am I saying?! I meant `<details>` is the one that was focusable, not `<summary>`. Gotcha on the `tabIndex` property.

MartijnCuppens commented:
Here's a screenshot of the focused `<summary>`, do you see something else?

![image](https://user-images.githubusercontent.com/11559216/83658191-29b22280-a5c2-11ea-9435-2a1788cb7aba.png)

zellwk commented:
My bad, just checked it properly with `document.activeElement` again. It's summary. Thanks for your patience and repeatedly pointing this out to me
